src/main/kotlin/org/rust/lang/core/psi/ext/PsiElement.kt

```diff
@@ -332,3 +332,20 @@ fun PsiElement.isKeywordLike(): Boolean {
         else -> false
     }
 }
+
+/**
+ * Consider we do some `resolve` in a quick-fix which is called in preview mode.
+ * Quick-fix is called on copy of the original file, but `resolve` will return original element.
+ * We will have an exception if we try to modify the original element.
+ * Thus, we should call this function on `resolve` result to obtain element in the copy of the original file.
+ */
+fun <T: PsiElement> T.findPreviewCopy(startElement: PsiElement): T? {
```

Comment:
Maybe
```suggestion
fun <T: PsiElement> T.findPreviewCopyIfNeeded(startElement: PsiElement): T? {
```
?

Actually, we don't do anything for elements outside of preview